src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/Protocol.java

```diff
@@ -203,4 +278,79 @@ private JSONArray dataEntry(Row dataRow, Schema schema) {
         }
         return entry;
     }
-}
+
+    public void generateCursorId() {
```

Comment:
suggest to separate cursor related code from Protocol and add UT for that.

Reply:
Refactored cursor related code from `Protocol`.